crates/subspace-farmer/src/plot.rs

```diff
@@ -989,9 +992,13 @@ impl<T: PlotFile> PlotWorker<T> {
                     .try_into()
                     .expect("Failed to decode piece offsets from rocksdb"),
             );
-            iter.next();
+            let index = self.piece_offset_to_index.get_piece_index(offset)?;
+            if &PieceIndexHash::from(index) <= from {
+                break;
+            }
 
-            piece_indexes.push(self.piece_offset_to_index.get_piece_index(offset)?)
+            piece_indexes.push(index);
+            iter.next();
```

Comment:
Don't get what you mean here, but answered above.

Reply:
With above explanation I see what you're trying to do here, but I don't think the check is correct. You want piece index hashes smaller than `max`, not `from`. Otherwise it'll wrap around `max` into `min` until it reaches `from` from the other side.

These abstractions with layers of translation of indexes are tricky and you need to be really careful with them, debugging can be very hard when there is a bug like this.